packages/web-components/fast-foundation/src/utilities/property-stylesheet-behavior.ts

```diff
@@ -0,0 +1,72 @@
+import {
+    Behavior,
+    ElementStyles,
+    FASTElement,
+    Observable,
+} from "@microsoft/fast-element";
+
+/**
+ * A behavior to add or remove a stylesheet from an element based on a property. The behavior ensures that
+ * styles are applied while the property matches and that styles are not applied if the property does
+ * not match.
+ *
+ * @public
+ */
+export class PropertyStyleSheetBehavior implements Behavior {
+    /**
+     * Constructs a {@link PropertyStyleSheetBehavior} instance.
+     * @param propertyName - The property name to operate from.
+     * @param value - The property value to operate from.
+     * @param styles - The styles to coordinate with the property.
+     */
+    constructor(
+        private propertyName: string,
+        private value: any,
+        private styles: ElementStyles
+    ) {}
+
+    /**
+     * Binds the behavior to the element.
+     * @param elementInstance - The element for which the property is applied.
+     */
+    public bind(elementInstance: FASTElement) {
+        Observable.getNotifier(elementInstance).subscribe(this, this.propertyName);
+        this.handleChange;
+    }
+
+    /**
+     * Unbinds the behavior from the element.
+     * @param source - The element for which the behavior is unbinding.
+     * @internal
+     */
+    public unbind(source: typeof FASTElement & HTMLElement) {
+        Observable.getNotifier(source).unsubscribe(this, this.propertyName);
+        (source as any).$fastController.removeStyles(this.styles);
+    }
+
+    /**
+     * Change event for the provided element.
+     * @param source - the element for which to attach or detach styles.
+     * @internal
+     */
+    public handleChange(source: FASTElement, key) {
+        if (source[key] === this.value) {
+            source.$fastController.addStyles(this.styles);
+        } else {
+            source.$fastController.removeStyles(this.styles);
+        }
+    }
+}
+
+/**
```

Comment:
I'm wondering if the is the appropriate place to export `appearanceBehavior` from. Nothing in fast-foundation has an `appearance` property so I think I would expect this to be exported from fast-components

Reply:
I think that makes sense, @chrisdholt thoughts?

Reply:
I'd agree for "appearance" and would agree we need to do a one off in Fluent for this. I think that's okay. If Fluent decided to move away from "appearance" for some other semantic, we'd want to be sure to update this accordingly rather than inheriting those semantics.